src/rpc/blockchain.cpp

```diff
@@ -1654,12 +1654,25 @@ bool FillScRecordFromInfo(const uint256& scId, const CSidechain& info, CSidechai
         sc.pushKV("immatureAmounts", ia);
 
         UniValue sf(UniValue::VARR);
-        for(const auto& entry: info.scFees)
-        {
-            UniValue o(UniValue::VOBJ);
-            o.pushKV("forwardTxScFee", ValueFromAmount(entry.forwardTxScFee));
-            o.pushKV("mbtrTxScFee", ValueFromAmount(entry.mbtrTxScFee));
-            sf.push_back(o);
+        // v2 non-ceasable sc
+        if (info.isNonCeasing()) {
+            for(const auto& entry: info.scFees_v2)
+            {
+                UniValue o(UniValue::VOBJ);
+                o.pushKV("forwardTxScFee", ValueFromAmount(entry.forwardTxScFee));
+                o.pushKV("mbtrTxScFee", ValueFromAmount(entry.mbtrTxScFee));
+                o.pushKV("submissionHeight", entry.submissionHeight);
+                sf.push_back(o);
```

Comment:
I'd suggest to 
```suggestion
                sf.push_back(std::move(o));
```
here, to save some mem copies.

Reply:
That would be a wise hint for a method on Univalue implementing `&&` in the 	`push_back` signature.
I fear it'd have no effect here.
(Just my two cents)

Reply:
Yes, I also changed all the occurrences in the same function in cb80200cd4274e56654dd8b0860c37dcb80cc076